### ISSUE TITLE
BE: Ändern einer Begegnung erzeugt Fehlermeldung

### ISSUE THREAD

**Issue by svdoldie** [diddipoeler/sportsmanagement#1269]
Wenn ich bei einer Begegnung z.B. das Datum oder die Uhrzeit ändere, bekomme ich beim speichern die angehängte Fehlermeldung.  
![Screenshot 2021-07-30 085220](https://user-images.githubusercontent.com/2417767/127613319-7aa3b47e-d550-49e0-afe1-e1d593e6689e.png)

**Comment by diddipoeler** [comment]
Backend oder Frontend ?

**Comment by svdoldie** [comment]
Hast du schon eine Idee woran das liegen kann?

**Comment by svdoldie** [comment]
Im Backend "matches", im Frontend results
Im Frontend gibts auch noch ein Problem mit dem Datum,
siehe Bilder. Beide sind vom selbem Projekt und Spieltag
![Anmerkung 2021-10-14 102214](https://user-images.githubusercontent.com/2417767/137279702-ebd48241-27bb-47d1-930b-35e70b709444.jpg)
![Anmerkung 2021-10-14 102316](https://user-images.githubusercontent.com/2417767/137279740-78367718-f5b0-4533-aa20-9b52267f8538.jpg)

**Comment by svdoldie** [comment]
Ich habe gerade nochmal stichpunktartig geguckt. Bei mir sind in allen Projekten so merkwürdige Datumsangaben

**Comment by svdoldie** [comment]
Die falschen Datumsangaben im Frontend scheinen nur Joomla 3 zu betreffen

**Comment by ortwin20000** [comment]
Kann ich mir im Backend die Fehlermeldung bei dir anschauen? Müsste dafür den Debug-Modus aktivieren.

**Comment by svdoldie** [comment]
Wenn, dann habe ich das nicht gemacht. Das brauche ich auch nicht

**Comment by svdoldie** [comment]
Nee, das geht nicht, weil damit Einstellungen für mehrere Module verbunden sind. Da muss es eine andere Möglichkeit geben.

**Comment by ortwin20000** [comment]
Das weiß ich nicht. Die Funktion ist ja gut aber eine Prüfung fehlt noch und die baue ich dann ein.

**Comment by svdoldie** [comment]
Hört sich gut an :-)  👍

**Comment by ortwin20000** [comment]
Ich würde folgendes umsetzen:
![grafik](https://user-images.githubusercontent.com/15192502/137582618-0c725dfa-ddbc-4d7d-8b41-ff72520fb122.png)
Neue Option im Tab "Lieblingsteam", mit dieser man die Benachrichtigung steuern kann.
Drei Optionen zur Auswahl: Keine Benachrichtigung, alle Spiele oder nur eigene Spiele.

Dort wo derzeit der Mailversand stattfindet prüfe ich dann die eingestellte Option.
Zusätzlich werde ich Leerzeichen für den Empfänger entfernen.
In deinen Spielern ist keine E-Mail-Adresse eingetragen, trotzdem versucht er eines zu schicken. Dies wird bei mir nicht durchgeführt, daher vermute ich, dass bei einem Spieler vielleicht ein Leerzeichen enthalten ist.

Ist das so in Ordnung?

**Comment by svdoldie** [comment]
Super :-) 👍

**Comment by diddipoeler** [comment]
Datenbankänderungen nicht im Code vornehmen. Ich erstelle eine neue Version der Komponente und
dann wird das Feld automatisch in der Tabelle hinzugefügt. Ist dann ein bisschen mehr Aufwand.

**Comment by svdoldie** [comment]
Ich bekomme folgende Fehlermeldung: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ''#__sportsmanagement_project' ADD 'fav_team_send_mail' TINYINT(4) NOT NULL DEFAU' at line 1

**Comment by svdoldie** [comment]
Beim Versuch eine Begegnung zu ändern bekomme ich folgende Meldung:  Unknown column 'p.fav_team_send_mail' in 'field list'
Kann es an dem p. vorweg liegen?

**Comment by svdoldie** [comment]
Ich bin schon auf dem neuesten Stand

**Comment by ortwin20000** [comment]
Hast du nach dem Update die neue Spalte in der Tabelle project?

**Comment by ortwin20000** [comment]
@diddipoeler hat nach dem Update auf 3.8.86 noch eine Änderung am Sql-Statement durchgeführt. Das dürfte bei dir nicht ausgeführt worden sein. Kannst du es manuell ausführen?

**Comment by sigal** [comment]
Feld wurde nicht angelegt.
SQL-Syntax: [ALTER TABLE `xxxx_sportsmanagement_project` ADD `fav_team_send_mail` TINYINT(4) NOT NULL DEFAULT '1';](url)

xxxx mit Präfix der eigen ersetzen!
